tools/HiveCompare/HiveCompare/Program.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            cli.ShowHelp();
+        }
+    }
+
+    static CommandLineApplication CreateCommandLineInterface()
+    {
+        var cli = new CommandLineApplication
+        {
+            Name = "HiveCompare"
+        };
+        cli.HelpOption("-?|-h|--help");
+        var firstFileOption = cli.Option("-f|--first-file", "first file to be used for comparison", CommandOptionType.SingleValue);
+        var secondFileOption = cli.Option("-s|--second-file", "second file to be used for comparison", CommandOptionType.SingleValue);
+        cli.OnExecute(() =>
+        {
+            bool HasRequiredOption(CommandOption option, [NotNullWhen(false)]out int? code)
+            {
+                if (!option.HasValue() || string.IsNullOrEmpty(option.Value()))
+                {
+                    cli!.ShowHelp();
+                    code = 1;
+                    return false;
+                }
+                code = null;
+                return true;
+            }
+            bool RequiredFileExists(CommandOption option, [NotNullWhen(false)] out int? code)
+            {
+                if (!File.Exists(option.Value()))
+                {
+                    Console.WriteLine($"Could not find file '{option.Value()}'.");
+                    code = 0;
+                    return false;
+                }
+                code = null;
+                return true;
+            }
+            if (!HasRequiredOption(firstFileOption, out int? code))
+                return code.Value;
+            if (!HasRequiredOption(secondFileOption, out code))
+                return code.Value;
+            if (!RequiredFileExists(firstFileOption, out code))
+                return code.Value;
+            if (!RequiredFileExists(secondFileOption, out code))
+                return code.Value;
+            if (!ParseTests(firstFileOption.Value(), secondFileOption.Value(), out code))
+                return code.Value;
+            var builder = new ContainerBuilder();
+            builder.RegisterType<ConsoleLogger>().As<ILogger>();
+            builder.RegisterType<BookEngine>().As<IBookEngine>();
+            var container = builder.Build();
+            var bookEngine = container.Resolve<IBookEngine>();
+            bookEngine.Config = new BookEngineConfig()
+            {
+                FixArabicYeKe = fixArabicOption.HasValue(),
+                FixVirtualSpaceAndPrefixSuffixes = true,
+                PersianShape = !noOptOption.HasValue()
+            };
+            try
+            {
+                var book = bookEngine.ProcessBook(inputFile.Value());
+                var outputPath = outputFile.HasValue() ? outputFile.Value() : null;
+                var savedPath = book.SaveAs();
+                if (outputPath != null)
+                {
+                    File.Copy(savedPath, outputPath, true);
+                    File.Delete(savedPath);
+                    savedPath = Path.GetFullPath(outputPath);
+                }
```

Comment:
Why can't we target `outputPath` in the first place?

Reply:
Code removed